packages/nimble-components/src/wafer-map/data-types/WaferMapRenderingObject.ts

```diff
@@ -0,0 +1,18 @@
+import type { WaferMapDataType } from '../types';
+import type { WaferMapDie } from './WaferMapDie';
+import { WaferMapMetadata } from './WaferMapMetadata';
+
+/**
+ * WaferMapRenderingObject
+ */
+export class WaferMapRenderingObject {
+    public dice!: WaferMapDie[];
```

Comment:
There are a lot of non-null assertions being added in the files. Nimble uses strict null checks so that we can get strong type checks on values. We should try to avoid them where possible, one option would be to require the user to configure them in the constructor, ie:

```ts
export class WaferMapRenderingObject {
    constructor(
      public dice: WaferMapDie[],
      public metadata: WaferMapMetadata,
      public maxCharacters: number,
      public waferDataType: WaferMapDataType,
      public colorsScale?: string[]
    ) { }

    public get isEmpty(): boolean {
        return this.dice.length > 0;
    }
}
```

Avoid non-null assertions and leveraging TypeScripts typing will allow us to avoid unnecessary null checking like in the `isEmpty` method above.

Reply:
I updated, thank you for the suggestion, I didn't know this aspect :)